File: src/main/java/com/qlm/zombie/dayphase/DayPhase.java
package com.qlm.zombie.dayphase;

import com.qlm.zombie.config.QLMConfig;
import net.minecraft.world.Difficulty;

public enum DayPhase {
    SAFE(Difficulty.PEACEFUL, "安全日"),
    EASY(Difficulty.EASY, "简单"),
    NORMAL(Difficulty.NORMAL, "普通"),
    HARD(Difficulty.HARD, "困难(锁定)"),
    EXTREME(Difficulty.HARD, "极限(锁定)");

    private final Difficulty difficulty;
    private final String displayName;

    DayPhase(Difficulty difficulty, String displayName) {
        this.difficulty = difficulty;
        this.displayName = displayName;
    }

    public static DayPhase forDay(long day) {
        int peacefulDays = QLMConfig.PEACEFUL_DAYS.get();
        int normalDays = QLMConfig.NORMAL_DAYS.get();
        int hardDays = QLMConfig.HARD_DAYS.get();
        int extremeDays = QLMConfig.EXTREME_DAYS.get();

        if (day <= peacefulDays) return SAFE;
        if (day <= normalDays) return EASY;
        if (day <= hardDays) return NORMAL;
        if (day <= extremeDays) return HARD;
        return EXTREME;
    }

    public int minDay() {
        int peacefulDays = QLMConfig.PEACEFUL_DAYS.get();
        int normalDays = QLMConfig.NORMAL_DAYS.get();
        int hardDays = QLMConfig.HARD_DAYS.get();
        int extremeDays = QLMConfig.EXTREME_DAYS.get();

        switch (this) {
            case SAFE: return 1;
            case EASY: return peacefulDays + 1;
            case NORMAL: return normalDays + 1;
            case HARD: return hardDays + 1;
            case EXTREME: return extremeDays + 1;
            default: return 1;
        }
    }

    public int maxDay() {
        int peacefulDays = QLMConfig.PEACEFUL_DAYS.get();
        int normalDays = QLMConfig.NORMAL_DAYS.get();
        int hardDays = QLMConfig.HARD_DAYS.get();
        int extremeDays = QLMConfig.EXTREME_DAYS.get();

        switch (this) {
            case SAFE: return peacefulDays;
            case EASY: return normalDays;
            case NORMAL: return hardDays;
            case HARD: return extremeDays;
            case EXTREME: return Integer.MAX_VALUE;
            default: return 1;
        }
    }

    public Difficulty difficulty() { return difficulty; }
    public String displayName() { return displayName; }

    public boolean isLocked() {
        return this == HARD || this == EXTREME;
    }
}

File: src/main/kotlin/com/qlm/zombie/dayphase/DayPhase.kt
package com.qlm.zombie.dayphase

enum class DayPhase(val minDay: Int, val maxDay: Int, val multiplier: Double, @get:JvmName("displayName") val displayName: String) {
    /** 0-25 天：和平 */
    PEACE(0, 25, 0.5, "和平"),
    /** 26-50 天：简单 */
    EASY(26, 50, 1.0, "简单"),
    /** 51-75 天：普通 */
    NORMAL(51, 75, 1.5, "普通"),
    /** 76-100 天：困难 */
    HARD(76, 100, 2.0, "困难"),
    /** 100 天+：锁定困难，无法更改 */
    LOCKED_HARD(101, Int.MAX_VALUE, 2.0, "锁定困难");

    val difficultyMultiplier: Double get() = multiplier

    fun isLocked(): Boolean = this == LOCKED_HARD

    /** 是否已进入最终锁定阶段（100 天后） */
    fun isFinal(): Boolean = this == LOCKED_HARD

    companion object {
        @JvmStatic
        fun fromDay(day: Int): DayPhase = entries.firstOrNull { day in it.minDay..it.maxDay } ?: LOCKED_HARD

        @JvmStatic
        fun forDay(day: Long): DayPhase = fromDay(day.toInt())
    }
}
